### ISSUE TITLE
Training Model Struggling with Character Recognition in Custom Sports Fonts

### ISSUE THREAD

**Issue by Suriyapongmax** [JaidedAI/EasyOCR#1253]
I am working on an OCR project aimed at accurately reading player numbers and names from sports images. These images feature 10 different custom fonts, predominantly thick and bold, which cater to a sports aesthetic. The primary challenge is the model's ability to distinguish between similar characters, particularly under the constraints of these stylized fonts.

Fonts: 10 custom sports fonts (English A-Z, a-z, 0-9).
Training Data: Generated dataset of ~200K images, including mixed cases of 3-10 characters with stroke and non-stroke and numbers (00-99) for each font.

After I train with num_iter: 750000 ,  loss: 0.00126, Valid loss: 0.15052

**Problems Encountered:**
- I test with Validate the accuracy is around 94%  but if I test with train set the accuracy is still 85-90%  which I think it should more than that because I train with 750K iterations and it should higher than Validate.
- Misidentification between numbers and letters (e.g., '10' read as '1O', '2' as 'Z').
- Confusion between similar characters (e.g., 'NATALIE' read as 'NATALlE' where 'I' and 'l' are confused; similar issues with 'Q' and 'O', 'Z' and 'z').
- Random incorrect predictions with no apparent pattern (e.g., 'EGH' to 'DF').

**Request for Help:**
I am seeking advice on improving my model’s performance in differentiating similar-looking characters,. Any suggestions on training strategies, network adjustments, or data preprocessing techniques would be greatly appreciated. 

here are acutal image I want to predict  
<img width="97" alt="Adihaus Bold_nostroke (7)" src="https://github.com/JaidedAI/EasyOCR/assets/32792068/2e4893b3-792b-47ea-a26e-47998098fdd8">
<img width="104" alt="Adihaus Bold_nostroke123 (12)" src="https://github.com/JaidedAI/EasyOCR/assets/32792068/598645cf-6358-4444-b699-906f85b47f29">
<img width="104" alt="Adihaus Bold_stroke123 (10)" src="https://github.com/JaidedAI/EasyOCR/assets/32792068/66d58e6b-c171-4bc6-a1c3-ef422b44b908">
<img width="101" alt="Alexandria_nostroke123 (29)" src="https://github.com/JaidedAI/EasyOCR/assets/32792068/deb1ce36-d925-4b98-b6b2-ae6e1fb6fdf4">

here are generated image I used as training set. 
![nbv5231665978](https://github.com/JaidedAI/EasyOCR/assets/32792068/86fcf8fc-a90f-4921-9223-d13a68a9765b)
![oc210030322606](https://github.com/JaidedAI/EasyOCR/assets/32792068/27df75bc-3185-4a86-83b6-6648d9370aa7)
![oc220080106966](https://github.com/JaidedAI/EasyOCR/assets/32792068/eface0ff-a45a-4b59-8ec8-0fb3a6edddf0)
![oc220080146883](https://github.com/JaidedAI/EasyOCR/assets/32792068/e9c5af9a-1662-4740-a3ad-36750ab22ed0)
![oc556143399804](https://github.com/JaidedAI/EasyOCR/assets/32792068/7f1f75f7-fd6d-4af1-9bd3-f639360f8c36)

**here is the config I use**
batch_size: 32

FT: False
optim: False
lr: 1
beta1: 0.9
total_data_usage_ratio: 1.0
batch_max_length: 34
imgH: 64
imgW: 600
rgb: False
contrast_adjust: 0.0
sensitive: True
PAD: True
data_filtering_off: False
Transformation: None
FeatureExtraction: VGG
SequenceModeling: BiLSTM
Prediction: CTC
num_fiducial: 20
input_channel: 1
output_channel: 256
hidden_size: 256
decode: greedy

THANK YOU IN ADVANCE !!

**Comment by alikhalil98771** [comment]
Hi, I want to ask you how long it took you to fine-tune the model with this number of iterations and dataset size, as I only used 3000 iterations and 20K images, and it is taking a very long time.

**Comment by Suriyapongmax** [comment]
> Hi, I want to ask you how long it took you to fine-tune the model with this number of iterations and dataset size, as I only used 3000 iterations and 20K images, and it is taking a very long time.

200K images with 40,000 Iteratios about 1 hour.

**Comment by alikhalil98771** [comment]
> > Hi, I want to ask you how long it took you to fine-tune the model with this number of iterations and dataset size, as I only used 3000 iterations and 20K images, and it is taking a very long time.
> 
> 200K images with 40,000 Iteratios about 1 hour.

Thanks for the reply. Also can you please tell me the gpu specification where I have 3050 12G and it still going more than 4 hours.

**Comment by Suriyapongmax** [comment]
> > > Hi, I want to ask you how long it took you to fine-tune the model with this number of iterations and dataset size, as I only used 3000 iterations and 20K images, and it is taking a very long time.
> > 
> > 
> > 200K images with 40,000 Iteratios about 1 hour.
> 
> Thanks for the reply. Also can you please tell me the gpu specification where I have 3050 12G and it still going more than 4 hours.

I use 4060Ti 16GB.  How much learning rate & Batch size you choose ?

**Comment by alikhalil98771** [comment]
> > > > Hi, I want to ask you how long it took you to fine-tune the model with this number of iterations and dataset size, as I only used 3000 iterations and 20K images, and it is taking a very long time.
> > > 
> > > 
> > > 200K images with 40,000 Iteratios about 1 hour.
> > 
> > 
> > Thanks for the reply. Also can you please tell me the gpu specification where I have 3050 12G and it still going more than 4 hours.
> 
> I use 4060Ti 16GB. How much learning rate & Batch size you choose ?

I am using the following config:
```
manualSeed: 1111
workers: 4
batch_size: 128 #32
num_iter: 3000
valInterval: 200
FT: False
optim: False # default is Adadelta
lr: 1.
beta1: 0.9
rho: 0.95
eps: 0.00000001
grad_clip: 5
#Data processing
select_data: 'e' # this is dataset folder in train_data
batch_ratio: '1' 
total_data_usage_ratio: 1.0
batch_max_length: 35 
imgH: 64
imgW: 600
rgb: False
contrast_adjust: False
sensitive: True
PAD: True
contrast_adjust: 0.0
data_filtering_off: False
# Model Architecture
Transformation: 'None'
FeatureExtraction: 'ResNet'
SequenceModeling: 'BiLSTM'
Prediction: 'CTC'
num_fiducial: 20
input_channel: 1
output_channel: 512
hidden_size: 512
decode: 'greedy'
new_prediction: False
freeze_FeatureFxtraction: False
freeze_SequenceModeling: False
```

**Comment by sunny-fun** [comment]
Hello, I want to know how to train my own model. I have some text that needs to be recognized, but I don't know the font names for these texts. I only have multiple images to be identified. Is it possible to train a model this way? Also, are there any tutorials on this topic? I want to learn, but I don't know where to start.

3Q😍
